src/main/java/de/kittybot/kittybot/database/SQL.java

```diff
@@ -54,60 +57,30 @@ public static void createTable(String table){
 		}
 	}
 
-	public static Connection getConnection() throws SQLException{
-		return dataSource.getConnection();
-	}
-
-	public static boolean execute(PreparedStatement preparedStatement){
-		try{
-			return preparedStatement.execute();
+	public static <T> T getProperty(String guildId, Field<T> field){
+		try(var ctx = SQL.getCtx()){
+			var res = ctx.select(field).from(GUILDS).where(GUILDS.GUILD_ID.eq(guildId)).fetchOne();
+			if(res != null){
+				return res.getValue(field);
+			}
 		}
 		catch(SQLException e){
-			LOG.error("Error executing prepared statement", e);
-		}
-		return false;
-	}
-
-	public static ResultSet executeWithResult(PreparedStatement preparedStatement){
-		try{
-			preparedStatement.executeUpdate();
-			return preparedStatement.getGeneratedKeys();
-		}
-		catch(SQLException e){
-			LOG.error("Error while executeWithResult prepared statement", e);
+			LOG.error("Error while getting key " + field.getName() + " from guild " + guildId, e);
 		}
 		return null;
 	}
 
-	public static int update(PreparedStatement preparedStatement){
-		try{
-			return preparedStatement.executeUpdate();
-		}
-		catch(SQLException e){
-			LOG.error("Error update prepared statement", e);
-		}
-		return -1;
-	}
-
-	public static boolean exists(PreparedStatement preparedStatement){
-		try{
-			var result = query(preparedStatement);
-			return result != null && result.next();
-		}
-		catch(SQLException e){
-			LOG.error("Error exists prepared statement", e);
-		}
-		return false;
+	public static DSLContext getCtx() throws SQLException{
+		return DSL.using(dataSource.getConnection(), SQLDialect.POSTGRES);
```

Comment:
what about closing the connection you pass here? don't you need to close it?

Reply:
doesn't the try with resource does that?
hmmm

Reply:
I guess it doesn't because the call is in a method right?

Reply:
`DSLContext` implements `AutoCloseable` so it can be used in `try-with-resources` but that's only applicable when you let jooq itself manage the connection, for example when using `DSL.using("sql", "username", password")` method, that's a resource. in this case, it's not a resource so it will just usually be garbage collected, though the connection will be still open. that means you should use `try-with-resources` in `getCtx()` itself and return the `DSLContext` using the connection from the twr block.